doc/adr/0013-marconi-query-interface.rst

```diff
@@ -0,0 +1,130 @@
+.. _marconi_indexer:
+
+ADR 13: Marconi Query Interface
+======================================================
+
+Date: 2022-10-27
+
+Author(s)
+---------
+
+Kayvan Kazeminejad <kayvan.kazeminejad@iohk.io>
+
+Status
+------
+
+Draft
+
+Context
+-------
+Marconi Query Interface is both an executable and a Haskell library built on top of `marconi`.  In its current alpha version, the interface provides a reporting capability for a bounded list of `Bech32 Shelley Era addresses <https://github.com/cardano-foundation/CIPs/tree/master/CIP-0019#shelley-addresses>`_ and their corresponding `UTXO` `TxOutRef`.  The executable package provides for an HTTP `JSON-RPC 2.0 <https://www.jsonrpc.org/specification>`_ on top of the marconi Haskell libraries.
+
+
+Scope
+-------
+
+The scope of this document is the marcoin query interface executable package built on top:
+
+- `marconi`
+- `marconi-query`
+- JSON-RPC 2.0
+- HTTP
+
+Caution
+^^^^^^
+
+This is a living document as the interface is under development
+
+
+Decision
+--------
+
+After soliciting a number of potential clients, we made the following architectural decisions:
+
+* Minimize changes to the marconi back-end
+* JSON-RPC 2.0 for non-Haskell users
+* Bech32 Shelley Era addresses are provided through CLI
+* The alpha release will scope to reporting on `TxOutRef` of `UTXO`s only
+
+These decision were an architectural compromise with the following  cons and pros:
+
+**Pros:**
+
+- Provide the most valued functionality first and fast
+- Iterate quickly to provide more functionality
+- Built on top of existing marconi code base with no or minimal impact to marconi for speed delivery
+- Reusability of existing `JSON` encoder/decoder for all entities of interest
+- Reasonable real-time query response time for the HTTP JSON-RPC service
+
+**Cons:**
+
+- `SQLite` will remain as marconi back-end database
+- Alpha version **will not** provide:
+  - Streaming services
+  - Notification services
+- Shelley Era addresses only
+
+
+Implications
+^^^^^^
+
+A critical consequence of these compromises in the alpha version, is that the query interface must be the sole owner of the SQLite `utxos` database. Any other connection to this SQLite database will result in undefined behavior of the software due to the concurrency nature of SQLite internals.
+
+Reports
+------------
+
+From a birds eye view, the query interface is a reporting capability for the indexed portion of the blockchain.  Our initial solution with the alpha version, provides for reports on the `TxOutRef` of a given address.
```

Comment:
"birds eye" in this context should be "bird's-eye" 

Also I suggest using the word "provides" instead of "is" here: 

"From a bird's-eye view, the query interface provides a reporting capability..."

You don't need the comma in the second sentence: 

"Our initial solution with the alpha version provides for..."